be/src/runtime/exec_env.cpp

```diff
@@ -154,6 +154,16 @@ Status ExecEnv::_init(const std::vector<StorePath>& store_paths) {
     _pipeline_prepare_pool =
             new PriorityThreadPool("pip_prepare", num_prepare_threads, config::pipeline_prepare_thread_pool_queue_size);
 
+    int num_sink_io_threads = config::pipeline_sink_io_thread_pool_thread_num;
+    if (num_sink_io_threads <= 0) {
+        num_sink_io_threads = std::thread::hardware_concurrency();
+    }
+    if (config::pipeline_sink_io_thread_pool_queue_size <= 0) {
+        return Status::InvalidArgument("pipeline_sink_io_thread_pool_queue_size shoule be greater than 0");
+    }
+    _pipeline_sink_io_pool =
```

Comment:
if the file sink DOP is only 1, by the default num_sink_io_threads, will we use many bthreads ( for many cpu cores ) to handle the file sink chunks?  is that reasonable?

Reply:
> if the file sink DOP is only 1, by the default num_sink_io_threads, will we use many bthreads ( for many cpu cores ) to handle the file sink chunks? is that reasonable?

1. for a single query, **only one** pthread will be used for file sink. the file chunks in the execution queue will be processed one by one in order.
2. these sink io threads will be shared amony all queries that contain sync io sinks(e.g. file sink/export sink).the thread pool is used to avoid creating threads frequently for each query.  the tasks in this thread pool are mainly used to perform sync IO operation without consuming a lot of CPU resources.